securesystemslib/formats.py

```diff
@@ -408,6 +374,71 @@ def _create_gpg_pubkey_with_subkey_schema(pubkey_schema):
     info = SCHEMA.Optional(SCHEMA.Any()),
   )
 
+# A single signature of an object.  Indicates the signature, and the KEYID of
+# the signing key.  I debated making the signature schema not contain the key
+# ID and instead have the signatures of a file be a dictionary with the key
+# being the keyid and the value being the signature schema without the keyid.
+# That would be under the argument that a key should only be able to sign a
+# file once.
+SIGNATURE_SCHEMA = SCHEMA.Object(
+  object_name = 'SIGNATURE_SCHEMA',
+  keyid = KEYID_SCHEMA,
+  sig = HEX_SCHEMA)
+
+# A schema holding the result of checking the signatures of a particular
+# 'SIGNABLE_SCHEMA' role.
+# For example, how many of the signatures for the 'Target' role are
+# valid?  This SCHEMA holds this information.  See 'sig.py' for
+# more information.
+SIGNATURESTATUS_SCHEMA = SCHEMA.Object(
+  object_name = 'SIGNATURESTATUS_SCHEMA',
+  threshold = SCHEMA.Integer(),
+  good_sigs = KEYIDS_SCHEMA,
+  bad_sigs = KEYIDS_SCHEMA,
+  unknown_sigs = KEYIDS_SCHEMA,
+  untrusted_sigs = KEYIDS_SCHEMA)
+
+# A dict where the dict keys hold a keyid and the dict values a key object.
+KEYDICT_SCHEMA = SCHEMA.DictOf(
+  key_schema = KEYID_SCHEMA,
+  value_schema = KEY_SCHEMA)
+
+ANY_SIGNATURE_SCHEMA = securesystemslib.schema.OneOf([SIGNATURE_SCHEMA,
+    GPG_SIGNATURE_SCHEMA])
+
+# List of ANY_SIGNATURE_SCHEMA.
+SIGNATURES_SCHEMA = SCHEMA.ListOf(ANY_SIGNATURE_SCHEMA)
+
+# A signable object.  Holds the signing role and its associated signatures.
+SIGNABLE_SCHEMA = SCHEMA.Object(
+  object_name = 'SIGNABLE_SCHEMA',
+  signed = SCHEMA.Any(),
+  signatures = SIGNATURES_SCHEMA)
+
+# Note: Verification keys can have private portions but in case of GPG we
+# only have a PUBKEY_SCHEMA (because we never export private gpg keys from
+# the gpg keyring)
+ANY_VERIFICATION_KEY_SCHEMA = SCHEMA.OneOf([ANYKEY_SCHEMA,
+    GPG_PUBKEY_SCHEMA])
+
+VERIFICATION_KEY_DICT_SCHEMA = SCHEMA.DictOf(
+  key_schema = KEYID_SCHEMA,
+  value_schema = ANY_VERIFICATION_KEY_SCHEMA)
+
+ANY_KEYDICT_SCHEMA = SCHEMA.OneOf([KEYDICT_SCHEMA,
+    VERIFICATION_KEY_DICT_SCHEMA])
+
+ANY_PUBKEY_SCHEMA = SCHEMA.OneOf([PUBLIC_KEY_SCHEMA, GPG_PUBKEY_SCHEMA])
+
+ANY_PUBKEY_DICT_SCHEMA = SCHEMA.DictOf(
+  key_schema = KEYID_SCHEMA,
+  value_schema = ANY_PUBKEY_SCHEMA)
+
+ANY_STRING_SCHEMA = SCHEMA.AnyString()
+LIST_OF_ANY_STRING_SCHEMA = SCHEMA.ListOf(ANY_STRING_SCHEMA)
```

Comment:
Any reason why `ANY_STRING_SCHEMA` and `LIST_OF_ANY_STRING_SCHEMA` were moved down here?

Reply:
It was the latest defined schema in the file, I moved it to the top of the file now.